### PR TITLE
Woche05 (1639024365;6260)

### PR DESCRIPTION
**Pull request by TomKnoden**
Lösungen zu den Aufgaben:
Nr.1) 1639024365
Nr.2)  6260

### CODE REVIEW

**Comment by paulseidel08** [comment]
Please write the outcomes of the exercises in the title of your pull request

**Comment by paulseidel08** [comment]
Don't put multiple Exercises in one pull request. e.g. the pull request of week 05 should only contain Exercise05

**Review by paulseidel08 — CHANGES_REQUESTED** [review]
Your Code for Exercises05 works totally fine and seems very efficient to me. Good Job!
Maybe you could add some docstrings next time, to make your code even more readable.

But for your code to be merged, you have to only put Exercise05 in your Pull Request. At the Moment, there are also the Exercises 04 and 03 included. For that I'm requesting changes. Also you should put the outcomes of your solutions in the title of your pull request.

File: Exercises/Tom_Knoden/Woche05/Woche_05_Aufgaben.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    valid = True
+                    break
+
+        if not valid:
+            return current_number
+
+    return None
+
+# Auslesen der Eingabe aus der Datei
+with open('data/input_sequence.txt', 'r') as file:
```
Comment (ClasherKasten) [inline Exercises/Tom_Knoden/Woche05/Woche_05_Aufgaben.py:20]: Not OS independent.

File: Exercises/Tom_Knoden/Woche05/Woche_05_Aufgaben.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            for content in contents:
+                if content == "no other bags.":
+                    continue
+                count, *color_parts = content.split()
+                color = " ".join(color_parts[:-1])
+                bag_rules[container].append((color, int(count)))
+    return bag_rules
+
+# Lesen der Eingabe aus der Datei
+bag_rules = parse_bag_rules('data/input_bags.txt')
```
Comment (ClasherKasten) [inline Exercises/Tom_Knoden/Woche05/Woche_05_Aufgaben.py:58]: Not OS independent.

**Review by ClasherKasten — CHANGES_REQUESTED** [review]
Overall good but missing type annotations.

File: Exercises/Tom_Knoden/Woche05/Woche_05_Aufgaben.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+#Aufgabenteil 1
+
+def find_first_invalid_number(numbers, preamble_length):
+    for i in range(preamble_length, len(numbers)):
+        current_number = numbers[i]
```
Comment (ClasherKasten) [inline Exercises/Tom_Knoden/Woche05/Woche_05_Aufgaben.py:5]: Can be more pythonic using `enumerate(...)`

**Review by ClasherKasten — CHANGES_REQUESTED** [review]

**Pull request closed without merging**